src/main/java/org/edgexfoundry/serviceactivator/MQTTOutboundServiceActivator.java

```diff
@@ -47,9 +48,31 @@ public String mqttOutbound(Message<?> msg) {
 			ExportString exportString = (ExportString) msg.getPayload();
 			logger.debug("message arrived at MQTT outbound sender: " + exportString.getEventId());
 			Addressable addressable = exportString.getRegistration().getAddressable();
-			if (addressable != null) {
+			if (addressable != null) {				
+				// if path contains "/" pre- or append path to topic, otherwise
+				// if path contains "\[any_string]" pre- or append dynamically device id to topic
+				// TODO - read "\[key]" from addressable and pre- or append its value to topic
+				String path = addressable.getPath();
+				if (path != null && !path.equals("")) {
+					String topic = addressable.getTopic();
+					if (path.contains("/")) {
```

Comment:
IMHO, both invocations of `contains` might be unnecessary, since in general they are more expensive than accessing individual characters.